myokit/formats/sbml/_importer.py

```diff
@@ -35,334 +36,1065 @@ def __init__(self):
         self.re_name = re.compile(r'^[a-zA-Z]+[a-zA-Z0-9_]*$')
         self.re_alpha = re.compile(r'[\W]+')
         self.re_white = re.compile(r'[ \t\f\n\r]+')
-        self.units = {}
 
-    def _convert_name(self, name):
+    def info(self):
+        return info
+
+    def supports_model(self):
         """
-        Converts a name to something acceptable to myokit.
+        Returns a bool indicating if model import is supported.
         """
-        if not self.re_name.match(name):
-            org_name = name
-            name = self.re_white.sub('_', name)
-            name = self.re_alpha.sub('_', name)
-            if not self.re_name.match(name):
-                name = 'x_' + name
-            self.logger().warn(
-                'Converting name <' + org_name + '> to <' + name + '>.')
-        return name
+        return True
 
-    def _convert_unit(self, unit):
+    def supports_component(self):
         """
-        Converts an SBML unit to a myokit one using the lookup tables generated
-        when parsing the XML file.
+        Returns a bool indicating if component import is supported.
         """
-        if unit in self.units:
-            return self.units[unit]
-        elif unit in unit_map:
-            return unit_map[unit]
-        else:      # pragma: no cover
-            raise SBMLError('Unit not recognized: ' + str(unit))
+        return True
 
-    def info(self):
-        return info
+    def parse_file(self, path):
```

Comment:
I like these methods, but it might be a good idea to have the following structure instead:

```
myokit.formats.sbml
   .SBMLParser
      .parse_string()
      etc
      does dirty work, gives the user fancy options like parsing strings
    .SBMLImporter
      .model()
      .supports_model()
      . etc. This thing just makes calls to an SBMLParser object, and doesn't give all the nice features, but instead provides the unified myokit.formats.Importer interface